demos/examples/bootstrap/src/Components/LayerList/hooks.ts

```diff
@@ -0,0 +1,62 @@
+import Handles from "@arcgis/core/core/Handles";
+import LayerListViewModel from "@arcgis/core/widgets/LayerList/LayerListViewModel";
+import { useEffect, useState } from "react";
+import { addLayerListHandlers } from "./utilities";
+import Collection from "@arcgis/core/core/Collection";
+import { CallBackData, LayerListViewModelState } from "./interfaces";
+
+export function useLayerListViewModel(
+  props: __esri.LayerListViewModelProperties
+) {
+  const { view } = props;
+  const [layerListVM, setLayerListVM] = useState<LayerListViewModel>(null);
+  const [state, updateState] = useState<LayerListViewModelState>(null);
+  const [operationalItems, updateOperationalItems] = useState<
+    __esri.Collection<__esri.ListItem>
+  >(new Collection());
+
+  useEffect(() => {
+    const handles = new Handles();
+
+    addLayerListHandlers({
+      layerListVM,
+      handles,
+      callback: (data: CallBackData) => {
```

Comment:
Looks like this callback is different correct? Can we pass this in or make it shared?

Reply:
Yeah - I created seperate files so we can compare the two. But I think we can delete the old one, add this to a shared folder, and update both demos to use the new version